test/simulator/misc/critical.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe('Critical hits', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should not happen on self-hits', function () {
+		battle = common.createBattle();
+		battle.join('p1', 'Guest 1', 1, [{species: "Zubat", moves: ['confuseray']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: "Chansey", item: 'luckypunch', moves: ['softboiled']}]);
+		for (let i = 0; i < 16; i++) {
+			battle.makeChoices('move confuseray', 'move softboiled');
```

Comment:
Just use Focus Energy first, instead of retrying 16 times.

Reply:
Actually, just give it the ability Super Luck; save a turn.

Reply:
Hm, there is a bit of risk that the Pokemon won't hit yourself due to confusion, this is random.

Reply:
Oh, I forgot that was one of the things you were testing.

I think there are seeds that guarantee crits here.